### ISSUE TITLE
"Error: Not permitted. Untrusted code may only update documents by ID. [403]"

### ISSUE THREAD

**Issue by leoetlino** [loneleeandroo/ngMeteor#38]
meteorCollection.save() on version 0.2 currently fails with the below error:

> Error: Not permitted. Untrusted code may only update documents by ID. [403]

It seems that ngMeteor is updating the collection:

```
collection.update(item_id, {$set: item}, function (error) {
```

However, that does not seem to be enough. Meteor will still deny ngMeteor from updating the collection so it won't work. If I change that line to:

```
collection.update(new Meteor.Collection.ObjectID(item_id._str), {$set: item}, function (error) {
```

...which is what I use when finding one element from a collection given an object ID, then it works as expected and the collection is updated correctly.

Is _this_ the solution or am I using ngMeteor wrong? Wondering if others are having the issue.

**Comment by leoetlino** [comment]
Fixed in 327a757 in my fork. Waiting on feedback before doing a pull request.

**Comment by Urigo** [comment]
Hi @leoetlino , after trying to update ngMeteor with pull requests for a while, it looks like ngMeteor is not really supported...
So i've forked ngMeteor and added support to Meteor 0.8.3 to the 0.2 branch.
I've also pull requested it but still no response.
So I guess me and my team will support our fork from now on.
Here it is:
https://github.com/Urigo/ngMeteor/tree/v0.2

I would love for you to test your error on our repo and if it is still happening create your pull request there, we would love to merge it!

In a week or so we will merge 0.2 branch to master on our repo.

**Comment by leoetlino** [comment]
I guess the author is really busy. So yes, it might be good to start a fork (this is what I've done currently).

**Comment by leoetlino** [comment]
As far as I can see from the commit, the code that's causing the issue hasn't been changed, so theroretically the error will still be there.

It's fixed in 327a7576d18a2cbb8afd103d8e7d6b28a6a0afae in my fork but I haven't tested it extensively, so I don't know if that breaks something or not.

**Comment by Urigo** [comment]
@leoetlino Good, so it means we need your fork on our fork :)
If you can fork our fork we will test it in our team and merge it as soon as possible.
We could change it ourselves but I would rather that you will get the credit for that pull request!

**Comment by Urigo** [comment]
Merged into https://github.com/Urigo/ngMeteor/tree/v0.2

**Comment by Urigo** [comment]
Now merged into master and released in the new release:
https://github.com/Urigo/ngmeteor/releases
